internal/driver/readhandler.go

```diff
@@ -24,23 +46,86 @@ func (d *Driver) HandleReadCommands(deviceName string, protocols map[string]mode
 
 	d.Logger.Debugf("Driver.HandleReadCommands: protocols: %v resource: %v attributes: %v", protocols, reqs[0].DeviceResourceName, reqs[0].Attributes)
 
-	// create device client and open connection
-	endpoint, err := config.FetchEndpoint(protocols)
+	endpoints, err := opcua.GetEndpoints(d.serviceConfig.OPCUAServer.Endpoint)
 	if err != nil {
-		return nil, err
```

Comment:
I think it makes more sense to log the error and return nil if there's an error rather than continuing

Reply:
Me too would prefer to `return nil, err` as it originally was if no important reasons are against it. Logging the error before returning is of course a good idea.

I think basically a _"fail fast"_ error handling strategy is better than continuation with _"best effort"_ unless something important is still to do like cleanup or so. By not returning the error the caller has no chance to detect if the function completed successfully or not.

In addition just logging and ignoring the error will probably leave `availableServerEndpoints` being `nil`. And `availableServerEndpoints` is further used below which could lead to exceptions which will get tinkered with `if availableServerEndpoints != nil` in turn and so on.

Reply:
done